openedx/features/enterprise_support/api.py

```diff
@@ -102,11 +97,12 @@ class EnterpriseApiClient(object):
     Class for producing an Enterprise service API client.
     """
 
-    def __init__(self):
+    def __init__(self, user=None):
```

Comment:
I think it would be better to create a new subclass of `EnterpriseApiClient` called `EnterpriseApiServiceClient` instead of defaulting to the service user here. See the [discovery clients](https://github.com/edx/edx-enterprise/blob/master/enterprise/api_client/discovery.py#L402) we have in edx-enterprise for an example.

Having separate classes makes it a little more obvious which type of user is going to be used to communicate with other services.